presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergCommonModule.java

```diff
@@ -200,6 +202,21 @@ public StatisticsFileCache createStatisticsFileCache(IcebergConfig config, MBean
         return new StatisticsFileCache(delegate);
     }
 
+    @Singleton
+    @Provides
+    public ManifestFileCache createManifestFileCache(IcebergConfig config, MBeanExporter exporter)
+    {
+        Cache<ManifestFileCacheKey, ManifestFileCachedContent> delegate = CacheBuilder.newBuilder()
+                .maximumWeight(config.getManifestCachingEnabled() ? config.getMaxManifestCacheSize() : 0)
```

Comment:
If the caching is disabled i think we should not have any caching instead of adding this via 0 weight

Reply:
I have updated the PR to remove use of the 0 weight and bypass the cache entirely if it is not enabled.